frontends/web/src/routes/account/summary/accountssummary.tsx

```diff
@@ -183,231 +106,144 @@ class AccountsSummary extends Component<Props, State> {
-                      </td>
-                    </tr>
-                  </tfoot>
-                </table>
-              </div>
-            </div>
+  // update the timer to get a new account summary update when receiving the previous call result.
+  useEffect(() => {
+    // replace previous timer if present
+    if (summaryReqTimerID.current) {
+      window.clearTimeout(summaryReqTimerID.current);
+    }
+    // set new timer
+    const delay = (!summaryData || summaryData.chartDataMissing) ? 1000 : 10000;
+    summaryReqTimerID.current = window.setTimeout(getAccountSummary, delay);
+  }, [summaryData]);
+
+  // update subscriptions on account change.
+  useEffect(() => {
+    subscribe();
+  }, [accounts, subscribe]);
+
+  // clear timer and subcription when unmounting the component.
+  useEffect(() => {
+    const currentUnsubscribeList = unsubscribeList.current;
+    return () => {
+      window.clearTimeout(summaryReqTimerID.current);
+      unsubscribe(currentUnsubscribeList);
+    };
+  }, []);
+
+  return (
+    <div className="contentWithGuide">
+      <div className="container">
+        <div className="innerContainer scrollableContainer">
+          <Status hidden={!hasCard} type="warning">
+            {t('warning.sdcard')}
+          </Status>
+          <Header title={<h2>{t('accountSummary.title')}</h2>}>
+            { debug && (
+              exported ? (
+                <A key="open" href="#" onClick={() => apiPost('open', exported)} title={exported} className="flex flex-row flex-start flex-items-center">
+                  <span>
+                    <Check style={{ marginRight: '5px !important' }} />
+                    <span>{t('account.openFile')}</span>
+                  </span>
+                </A>
+              ) : (
+                <a key="export" onClick={exportSummary} title={t('accountSummary.exportSummary')}>
+                  <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#699ec6" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
+                    <path d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"></path>
+                    <polyline points="7 10 12 15 17 10"></polyline>
+                    <line x1="12" y1="15" x2="12" y2="3"></line>
+                  </svg>
+                </a>
+              )
+            )}
+          </Header>
+          <div className="content padded">
+            <Chart
+              data={summaryData}
+              noDataPlaceholder={
+                (accounts.length === Object.keys(balances || {}).length) ? (
+                  <AddBuyReceiveOnEmptyBalances balances={balances} />
+                ) : undefined
+              } />
+            <SummaryBalance
+              accounts={accounts}
+              summaryData={summaryData}
+              totalBalancePerCoin={totalBalancePerCoin}
+              balances={balances}
+              syncStatus={syncStatus}
```

Comment:
fine for now, but it looks like all syncStatus related logic could be moved to SummaryBalance in the future.